py_proxy/views.py

```diff
@@ -70,22 +74,26 @@ def content_type(request):
     )
 
     with requests.get(url, stream=True, allow_redirects=True) as rsp:
-        if rsp.headers.get("Content-Type") in ("application/x-pdf", "application/pdf",):
+        if rsp.headers.get("Content-Type") in ("application/x-pdf", "application/pdf"):
             return exc.HTTPFound(
                 request.route_url(
-                    "pdf", pdf_url=request.matchdict["url"], _query=request.params
-                )
+                    "pdf", pdf_url=request.matchdict["url"], _query=request.params,
+                ),
+                headers=_caching_headers(max_age=300),
```

Comment:
What happens if the response from `requests.get()` is an unsuccessful response? We're asking requests to follow redirects so I think 3xx's are out. But what if we receive a 4xx or 5xx response (or a 1xx, which would be weird)? These would seem to indicate that something went wrong, so we probably wouldn't want to cache them.

Could be dealt with as a separate issue

Reply:
See code below?